lib/react.js

```diff
@@ -250,6 +250,12 @@ declare module react {
   declare export var ConcurrentMode: ({children: ?React$Node}) => React$Node; // 16.7+
   declare export var StrictMode: ({children: ?React$Node}) => React$Node;
 
+  declare export var Suspense: React$ComponentType<{
+    children?: ?React$Node,
+    fallback: React$Node,
+    maxDuration?: number
+  }>; // 16.6+
```

Comment:
Does this imply that this is a class? I guess that isn't technically correct since it's one of those internal special symbols like Fragment but maybe good enough for now.

Should it use the same hack as Fragments and pretend to be a functional component? That way maybe refs are invalid maybe?

Reply:
> Should it use the same hack as Fragments and pretend to be a functional component? That way maybe refs are invalid maybe?

Then people might think they can call it as a function, which would error?

Reply:
ComponentType is the type alias that (currently) represents function components and class components.

Reply:
Is there an alias for something that more accurately represents `StrictMode`, `Suspense`, etc?

Reply:
Not yet, but AbstractComponent will fulfill that role!

Reply:
Is React$AbstractComponent made for this?

Reply:
AbstractComponent was made to represent components more abstractly than we do today. When it's done, it'll be able to represent this, the component returned from forwardRef, and hopefully many more components that React releases in the future.

It's still a WIP, but I will release a blog post when it's fully landed in master.

Reply:
> Then people might think they can call it as a function, which would error? (Doesn't seem like that would obviously be better than this.)

If the instance type is void, it's more precise to use the function hack. Since we're already using that hack today, and since it will be fixed as part of my work on AbstractComponent, I think we should use the function representation for now.

Reply:
Ok~

Reply:
I must be missing something, but changing it to a function declaration seems break the validation.

For example, [this test](https://flow.org/try/#0JYWwDg9gTgLgBAJQKYEMDG8BmUIjgcilQ3wG4AocgEyTQBsUi4kAPSWOAN0bgGUBXAM5gkAO0FIAXImIwAJAGFckUWJgAVAJ4iAPAG9ycOGgAWwOlSKiA-NOvJ08gHIQaAGkNxMKOnQBG6ADWtjKOci7uniAoLAAi-FAoMMAQNtKi-CB+SFDkAL4AfKRwAPQlcACMAGwAdFUA1JQGRjoCwmIScNFxCUkpogC8AEQofmhDpQWl5QCiUDhQ0oIwUMCiAOZwwIJbomjKfX50SHAA7sAwJnAZWTn55EA) fails, as I would expect.

But [this one](https://flow.org/try/#0JYWwDg9gTgLgBAJQKYEMDG8BmUIjgcilQ3wG4AocgEyTQBsUi4kAPSWOAN0bgGUBXAM5gkAO0FIAXHAAUAb3Jw4aABbA6VIqID807cnQwAJADkINADSK4mFHToAjdAGtdiYsbOXrIFCwAi-FAoMMAQOtKi-CAOSFDkAL4AlHAAvAB87oam5kikcAD0BXAAjABsAHRlANSK5ApKADwCwmIScL4BQSFhoqkARCgOaP2FmUVwAKJQOFDSgjBQwKIA5nDAguuiaLhgPQ50SHAA7sAwKnBRMXGJ5EA) and [this one](https://flow.org/try/#0JYWwDg9gTgLgBAJQKYEMDG8BmUIjgcilQ3wG4AocgEyTQBsUi4kAPSWOTAVwDsNgIPOAGUuAZzBIeYpHAAUAb3Jw4aABbA6VIjwD8ALji7k6GABIAchBoAaZZxR06AI3QBrA4mLmrt+yBQWABEuKBQYAT1DHi4QZyQocgBfAEpDEwxLayRSOAB6PLgARgA2ADoSgGpKJRUAHlEJKRk4AODQ8MiAXgAiFGc0HvyAPnzCgFEoHChDMRgoYB4AczhgMVW+XDBO5zpZAHdgGDU4GLiE5PIgA) do not.

Reply:
Calling `Suspense` directly (as a plain function) e.g. `Suspense({ maxDuration: 'abc' })` properly catches the error, but wrapping it in a `createElement` call does not.